src/auth/README.md

```diff
@@ -72,69 +83,196 @@ if (blockstack.isUserSignedIn()) {
+  "start_url": "http://blockstack-todos.appartisan.com",
+  "description": "A simple todo app build on blockstack",
+  "icons": [{
+    "src": "http://blockstack-todos.appartisan.com/logo.png",
+    "sizes": "400x400",
+    "type": "image/png"
+  }]
+}
 ```
 
-### Verify an auth request
+The manifest file *MUST* have [Cross-origin resource sharing (CORS) headers](https://en.wikipedia.org/wiki/Cross-origin_resource_sharing) that allow the manifest file to be fetched from any arbitrary source. This usually means returning  
 
-```es6
-import { verifyAuthRequest } from 'blockstack'
+## Key pairs
 
-const verified = verifyAuthRequest(authRequest)
-```
+Blockstack Authentication makes extensive use of public key cryptography. As mentioned above, we use ECDSA with the secp256k1 curve. What follows is a description of the various public-private key pairs used in the authentication process including how they're generated, where they're used and to whom the private key is disclosed.
+
+### Transit private key
 
-### Create an auth response
+The transit private is an ephemeral key that is used to encrypt secrets that need to be passed from the Blockstack Browser to the app during the authentication process. It is randomly generate by the app at the beginning of the authentication response. The public key that corresponds to the transit private key is stored in a single element array in the `public_keys` key of the authentication request token. The Blockstack Browser encrypts secret data such as the app private using this public key and sends it back to the app when the user signs in to the app. The transit private key signs the app authentication request.
 
-```es6
-import { makeAuthResponse, makeECPrivateKey } from 'blockstack'
+### Blockstack ID Identity address private key
+The identity address private key is derived from the user's keychain phrase and is the private key of the Blockstack ID that the user chooses to use to sign in to the app. It is a secret owned by the user and never leaves the user's instance of the Blockstack browser. This private key signs the authentication response token for an app to indicate that the user approves sign in to that app.
 
-const privateKey = makeECPrivateKey()
-const profile = { "name": "Naval Ravikant" }
-const username = "naval.id"
-const authResponse = makeAuthResponse(privateKey, profile, username)
+### App private key
+
+The app private key is an app-specific private key that is generated from the user's identity address private key using the `domain_name` as input. It is deterministic in that for a given Blockstack ID and `domain_name`, the same private key will be generated each time. The app private key is securely shared with the app on each authentication, encrypted by the Blockstack browser with the transit public key.  
+
+The app private key serves three functions. 
+* It is used to create the credentials that give an app access to the gaia hub storage bucket for that specific app. 
```

Comment:
"It is used to create the credentials that give an app access to the gaia hub storage bucket for that specific app." ->
"It is used to create credentials for accessing the user's app-specific gaia storage bucket."

Also, this list can be numbers instead of bullets.